### PR TITLE
feat: web based applications will now flush events before closing to ensure events are sent

### PR DESCRIPTION
**Pull request by tanderson-ld**
**Requirements**

- [x] I have added test coverage for new or changed functionality

- [x] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)

- [x] I have validated my changes against all supported platform versions

**Describe the solution you've provided**

Created LDAppLifecycleListener and configured to use conditional imports.  In the web case, uses an LDAppLifecycleListener that uses html document visibility.  In the non-web case, uses the existing Flutter AppLifecycleListener.  Note that the web based LDAppLifecycleListener only emits resume and hidden, no other events.

### CODE REVIEW

File: packages/flutter_client_sdk/lib/src/connection_manager.dart
Comment (tanderson-ld) [inline packages/flutter_client_sdk/lib/src/connection_manager.dart:182]: Do not merge until bugfix for this line is merged separately to ensure changelog recording.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**
